server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1664,7 +1664,7 @@ public boolean isSegmentReplicationAllowed() {
             return false;
         }
         if (getReplicationEngine().isEmpty()) {
-            logger.warn(
+            logger.trace(
```

Comment:
nit: Is it possible to get into this state i.e. where engine is not defined or it is not `NRTReplicationEngine` ?

Reply:
Thanks @mch2. Yes, we can take this in a separate change. Please feel free to add a `todo` comment or create a tracking issue.